### PR TITLE
Message Component

### PR DESCRIPTION
**Pull request by asaloff**
This PR adds the Message component with docs.

### CODE REVIEW

File: src/components/message/index.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import { nanoid } from "nanoid";
+
+if (
+  typeof document !== "undefined" &&
+  !document.getElementById("svant-messages-wrapper")
+) {
+  // Create the component if it doesn't exist
+  new Message({ target: document.body });
+
+  const createMessageFunction = type => options => {
```
Comment (kujtimiihoxha) [inline src/components/message/index.js:13]: This function does seem really complicated and it is hard to follow.

Maybe one thing we could do is think of how we could split this up a bit, and also for the top level function (not the inner destroy function) we could maybe create a variable that stores the current `messageStore` so you don't have to always do `get(messageStore)`.

Comment (asaloff) [reply src/components/message/index.js:13]: Yup makes sense. It is a big function 😅. I'll break it up.

Comment (asaloff) [reply src/components/message/index.js:13]: I'll add some comments too.

Comment (asaloff) [reply src/components/message/index.js:13]: I ended up only adding comments. I think breaking it up would make it more complex. Also extracting `get(messageStore)` to a function was reading funny, so I left it as is.

https://github.com/svelte-components/svant/pull/21/commits/9e4e548110178312711a82a70e286f511ec0d12c
